hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientPartitionServiceImpl.java

```diff
@@ -231,10 +231,13 @@ public int getPartitionId() {
         @Override
         public Member getOwner() {
             Address owner = getPartitionOwner(partitionId);
-            if (owner != null) {
-                return client.getClientClusterService().getMember(owner);
+            if (owner == null) {
+                ClientMessage message = ClientTriggerPartitionAssignmentCodec.encodeRequest();
+                ClientInvocation invocation = new ClientInvocation(client, message, null);
+                invocation.invoke();
+                return null;
             }
-            return null;
+            return client.getClientClusterService().getMember(owner);
```

Comment:
what if partition assignment changed during the PartitionAssignment task. I think it may be better  to return `getPartitionOwner(partitionId);` but then what if this is null? It will return null but hopefully the api will eventually return non-null.